crypto/encode_decode/decoder_lib.c

```diff
@@ -322,10 +322,16 @@ int OSSL_DECODER_CTX_add_decoder(OSSL_DECODER_CTX *ctx, OSSL_DECODER *decoder)
     prov = OSSL_DECODER_get0_provider(decoder);
     provctx = OSSL_PROVIDER_get0_provider_ctx(prov);
 
-    if ((decoderctx = decoder->newctx(provctx)) == NULL
+    if (decoder->newctx_ex != NULL)
+        decoderctx = decoder->newctx_ex(provctx, NULL);
```

Comment:
Note I have not tried to address this NULL here.. It may need a OSSL_DECODER_CTX_add_decoder_ex() that passes a propq.
I did not want to do that here for backport reasons.